rte/kernels/mo_rte_solver_kernels.F90

```diff
@@ -947,5 +949,295 @@ pure subroutine apply_BC_0(ncol, nlay, ngpt, top_at_1, flux_dn) bind (C, name="a
+  ! Local variables
+  real(wp), dimension(ncol,nlay+1,ngpt) :: radn_dn, radn_up ! Fluxes per quad angle
+  real(wp), dimension(ncol,       ngpt) :: Ds_ncol
+
+  real(wp), dimension(ncol,nlay,  ngpt) :: tauLoc           ! rescaled Tau
+  real(wp), dimension(ncol,nlay,  ngpt) :: scaling          ! scaling
 
+  integer :: imu, top_level
+  real    :: weight
+  ! Tang rescaling
+  call scaling_1rescl(ncol, nlay, ngpt, tauLoc, scaling, tau, ssa, g) 	
+  ! ------------------------------------
+  !
+  ! For the first angle output arrays store total flux
+  !
+  top_level = MERGE(1, nlay+1, top_at_1)
+  Ds_ncol(:,:) = Ds(1)
+  weight = 2._wp*pi*weights(1)
+  radn_dn(1:ncol, top_level, 1:ngpt)  = flux_dn(1:ncol, top_level, 1:ngpt) / weight
+
+  call lw_solver_1rescl(ncol, nlay, ngpt, &
+                        top_at_1, Ds_ncol, tauLoc, scaling, &
+                        lay_source, lev_source_inc, lev_source_dec, sfc_emis, sfc_src, &
+                        flux_up, flux_dn)
+
+  flux_up = flux_up * weight
+  flux_dn = flux_dn * weight
+
+  do imu = 2, nmus
+    Ds_ncol(:,:) = Ds(imu)
+    weight = 2._wp*pi*weights(imu)
+    radn_dn(1:ncol, top_level, 1:ngpt)  = flux_dn(1:ncol, top_level, 1:ngpt) / weight
+    call lw_solver_1rescl(ncol, nlay, ngpt, &
+                          top_at_1, Ds_ncol, tauLoc, scaling, &
+                          lay_source, lev_source_inc, lev_source_dec, sfc_emis, sfc_src, &
+                          radn_up, radn_dn)
+
+    flux_up(:,:,:) = flux_up(:,:,:) + weight*radn_up(:,:,:)
+    flux_dn(:,:,:) = flux_dn(:,:,:) + weight*radn_dn(:,:,:)
+  end do
+  end subroutine lw_solver_1rescl_GaussQuad
+
+  pure subroutine scaling_1rescl(ncol, nlay, ngpt, tauLoc, scaling, tau, ssa, g)
+    integer ,                              intent(in)    :: ncol
+    integer ,                              intent(in)    :: nlay
+    integer ,                              intent(in)    :: ngpt
+    real(wp), dimension(ncol, nlay, ngpt), intent(in)    :: tau
+    real(wp), dimension(ncol, nlay, ngpt), intent(in)    :: ssa
+    real(wp), dimension(ncol, nlay, ngpt), intent(in)    :: g
+
+    real(wp), dimension(ncol, nlay, ngpt), intent(inout) :: tauLoc
+    real(wp), dimension(ncol, nlay, ngpt), intent(inout) :: scaling
+
+    integer  :: icol, ilay, igpt
+    real(wp) :: wb, ssal, scaleTau
+    !$acc enter data copyin(tau, ssa, g)
+    !$acc enter data create(tauLoc, scaling)
+    !$acc parallel loop collapse(3)
+    do igpt=1,ngpt
+      do ilay=1,nlay
+        do icol=1,ncol
+          ssal = ssa(icol, ilay, igpt)
+          wb = ssal*(1._wp - g(icol, ilay, igpt)) / 2._wp
+          scaleTau = (1._wp - ssal + wb )
+          ! Eq.15 of the paper
+          tauLoc(icol, ilay, igpt) = scaleTau * tau(icol, ilay, igpt)
+          ! 
+          ! here ssa is used to store parameter wb/[1-w(1-b)] of Eq.21 of the Tang's paper
```

Comment:
Please bring this comment up to date - there's no `ssa` variable used any more

Reply:
Done

Reply:
fixed